ext/mysql2/client.c

```diff
@@ -30,15 +30,17 @@ VALUE rb_hash_dup(VALUE other) {
     rb_raise(cMysql2Error, "MySQL client is not initialized"); \
   }
 
+#define CONNECTED(wrapper) (wrapper->client && wrapper->client->net.vio != NULL && wrapper->client->net.fd != -1)
```

Comment:
I think so, which is why I opened https://github.com/brianmario/mysql2/pull/796.  I will update that PR to use this macros after this gets merged.